Test/comp/Variance.dfy

```diff
@@ -0,0 +1,190 @@
+// RUN: %dafny /compile:3 /spillTargetCode:2 /compileTarget:cs "%s" > "%t"
+// RUN: %dafny /compile:3 /spillTargetCode:2 /compileTarget:js "%s" >> "%t"
+// RUN: %dafny /compile:3 /spillTargetCode:2 /compileTarget:go "%s" >> "%t"
+// RUN: %dafny /compile:3 /spillTargetCode:2 /compileTarget:java "%s" >> "%t"
+// RUN: %diff "%s.expect" "%t"
+// The Java compiler lacks support for this (see dafny0/RuntimeTypeTests0.dfy).
+
+datatype Co<+T> = Co(T){
+    const x : int;
+    const y : seq<T>;
+
+    function method A(x : T) : int { 0 }
+    static function method sA(x : T) : int { 0 }
+    function method gA(ghost x : T) : int { 0 }
+    function method B(x : seq<T>) : int { 0 }
+    function method C(x : int) : seq<T> { [] }
+    function method D(x : T) : T { x }
+
+    method mA(x : T) returns (r: int) { r := 0; }
+    method mD(x : T) returns (r: T) { r := x; }
+}
+
+datatype In<T> = In(T){
+    const x : int;
+    const y : seq<T>;
+
+    function method A(x : T) : int { 0 }
+    static function method sA(x : T) : int { 0 }
+    function method gA(ghost x : T) : int { 0 }
+    function method B(x : seq<T>) : int { 0 }
+    function method C(x : int) : seq<T> { [] }
+    function method D(x : T) : T { x }
+
+    method mA(x : T) returns (r: int) { r := 0; }
+    method mD(x : T) returns (r: T) { r := x; }
+}
+
+datatype Con<-T> = Con {
```

Comment:
Shouldn't there be an argument of the `Con` constructor that includes `T` ? 

Like `datatype Con<-T> = Con(T -> int)`